src/GovUk.Education.ExploreEducationStatistics.Admin/Services/Public.Data/DataSetService.cs

```diff
@@ -0,0 +1,119 @@
+                    .Include(ds => ds.Versions)
+                    .Where(ds => ds.PublicationId == publicationId);
+
+                var dataSets = (await dataSetsQueryable
+                        .OrderByDescending(ds => ds.LatestLiveVersion != null ? ds.LatestLiveVersion.Published : DateTimeOffset.MinValue)
+                        .ThenBy(ds => ds.Title)
+                        .ThenBy(ds => ds.Id)
+                        .Paginate(page: page, pageSize: pageSize)
+                        .ToListAsync(cancellationToken: cancellationToken)
+                    )
+                    .Select(MapDataSet)
+                    .ToList();
+
+                return new PaginatedListViewModel<DataSetViewModel>(
+                    dataSets,
+                    totalResults: await dataSetsQueryable.CountAsync(cancellationToken: cancellationToken),
+                    page: page,
+                    pageSize: pageSize);
+            });
+    }
+
+    private static DataSetViewModel MapDataSet(DataSet dataSet)
+    {
+        var latestDraftVersion = dataSet.Versions
+            .Where(version => version.Status is DataSetVersionStatus.Draft
+                or DataSetVersionStatus.Failed
+                or DataSetVersionStatus.Mapping
+                or DataSetVersionStatus.Processing)
+            .OrderBy(version => version.VersionMajor)
+            .ThenBy(version => version.VersionMinor)
+            .LastOrDefault();
+
+        return new DataSetViewModel
+        {
+            Id = dataSet.Id,
+            Title = dataSet.Title,
+            Summary = dataSet.Summary,
+            Status = dataSet.Status,
+            DraftVersion = MapDraftVersion(latestDraftVersion),
+            LatestLiveVersion = MapLiveVersion(dataSet.LatestLiveVersion),
+            SupersedingDataSetId = dataSet.SupersedingDataSetId,
+        };
+    }
+
+    private static DataSetVersionViewModel? MapDraftVersion(DataSetVersion? dataSetVersion)
+    {
+        return dataSetVersion != null
+            ? new DataSetVersionViewModel
+            {
+                Id = dataSetVersion.Id,
+                Number = dataSetVersion.Version,
+                Status = dataSetVersion.Status,
+                Type = dataSetVersion.VersionType,
+            }
+            : null;
+    }
+
+    private static DataSetLiveVersionViewModel? MapLiveVersion(DataSetVersion? dataSetVersion)
+    {
+        return dataSetVersion != null
+            ? new DataSetLiveVersionViewModel
+            {
+                Id = dataSetVersion.Id,
+                Number = dataSetVersion.Version,
+                Published = dataSetVersion.Published!.Value,
+                Status = dataSetVersion.Status,
+                Type = dataSetVersion.VersionType,
+            }
+            : null;
+    }
+
+    private async Task<Either<ActionResult, Publication>> CheckPublicationExists(Guid publicationId,
+        CancellationToken cancellationToken = default)
```

Comment:
No need to provide a default here as we'll always be guaranteed to be passing a `CancellationToken` through from the public method